addons/mail/static/tests/tours/mail_message_load_before_order_tour.js

```diff
@@ -0,0 +1,45 @@
+/* @odoo-module */
+
+import { registry } from "@web/core/registry";
+
+registry.category("web_tour.tours").add("mail_message_load_before_order_tour", {
+    test: true,
+    steps: () => [
+        {
+            trigger: ".o-mail-DiscussSidebarChannel:contains(MyTestChannel)",
+            run: "click",
+        },
+        {
+            trigger: ".o-mail-Message:contains(60)",
+            run() {
+                // Ensure messages 31-60 are loaded in order: the initial fetch
+                // retrieve the last 30 messages.
+                const messages = Array.from(
+                    document.querySelectorAll(".o-mail-Thread .o-mail-Message-content")
+                ).map((el) => el.innerText);
+                for (let i = 0; i < 30; i++) {
+                    if (messages[i] !== (30 + i + 1).toString()) {
+                        throw new Error("Wrong message order after initial load");
+                    }
+                }
+                const thread = document.querySelector(".o-mail-Thread");
+                thread.scrollTo({ top: 0 });
+            },
+        },
+        {
+            trigger: ".o-mail-Message:first:not(:contains(31))",
+            run() {
+                // Ensure messages 1-60 are loaded in order: scrolling to the
+                // top should load the previous 30 messages.
+                const messages = Array.from(
+                    document.querySelectorAll(".o-mail-Thread .o-mail-Message-content")
+                ).map((el) => el.innerText);
+                for (let i = 0; i < 60; i++) {
+                    if (messages[i] !== (i + 1).toString()) {
+                        throw new Error("Wrong message order after scrolling to the top");
+                    }
+                }
+            },
+        },
+    ],
+});
```

Comment:
I would prefer QUnit test rather than tour. This is very verbose and can bully future development, e.g. test being weak to implementation changes and/or test failing due to triggered RPCs after end of test (e.g. mark as read), or just that we don't usually run tours locally because it's slow, so it slows down quite a lot development if we have to wait runbot even more.

Reply:
RPC at the end of this tour would have no impact. QUnit might be easier to work with but its also less reliable (depending on mock server implementation and other mocks). I'd keep the tour for such an important test (message load is a very important discuss feature).